pkg/runner/expression.go

```diff
@@ -55,13 +55,23 @@ func (rc *RunContext) NewExpressionEvaluator() ExpressionEvaluator {
 		Needs:    using,
 		Inputs:   rc.Inputs,
 	}
-	return expressionEvaluator{
+
+	eV := expressionEvaluator{
 		interpreter: exprparser.NewInterpeter(ee, exprparser.Config{
 			Run:        rc.Run,
 			WorkingDir: rc.Config.Workdir,
 			Context:    "job",
 		}),
 	}
+	// evaluate environment variables since they can contain GitHub's special
+	// environment variables.
+	for k, v := range rc.GetEnv() {
+		valueEval, err := eV.evaluate(v, exprparser.DefaultStatusCheckNone)
+		if err == nil {
+			rc.Env[k] = fmt.Sprintf("%v", valueEval)
+		}
+	}
```

Comment:
The expression evaluator by itself should probably not update the RunContext.
It would be done many many times, a better place would be the preparation of the pipeline right before act exectues all jobs.

Reply:
> the preparation of the pipeline right before act exectues all jobs.
@KnisterPeter can you point me out where that is done?

Reply:
The RunContext is created here:
https://github.com/nektos/act/blob/859445fb94e12cc600e5a59d9c9cd687214cb801/pkg/runner/runner.go#L157

From there you can go through the code and look for an appropriate place to add your code.